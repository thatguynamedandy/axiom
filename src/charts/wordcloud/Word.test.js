import React from 'react';
import renderer from 'react-test-renderer';
import { Word } from 'bw-axiom';

function getComponent(props = { x: 0, y: 0 }) {
  return renderer.create(
    <Word { ...props } />
  );
}

describe('Word', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onClick', () => {
    const component = getComponent({ onClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    [
      'rose',
      'pink',
      'purple',
      'lilac',
      'blue',
      'teal',
      'green',
      'chartreuse',
      'amber',
      'orange',
      'brown',
      'grey',
    ].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
