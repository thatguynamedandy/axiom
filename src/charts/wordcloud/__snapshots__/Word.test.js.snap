// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word renders with color amber 1`] = `
<text
  className="ax-word ax-word--amber"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color blue 1`] = `
<text
  className="ax-word ax-word--blue"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color brown 1`] = `
<text
  className="ax-word ax-word--brown"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color chartreuse 1`] = `
<text
  className="ax-word ax-word--chartreuse"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color green 1`] = `
<text
  className="ax-word ax-word--green"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color grey 1`] = `
<text
  className="ax-word ax-word--grey"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color lilac 1`] = `
<text
  className="ax-word ax-word--lilac"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color orange 1`] = `
<text
  className="ax-word ax-word--orange"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color pink 1`] = `
<text
  className="ax-word ax-word--pink"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color purple 1`] = `
<text
  className="ax-word ax-word--purple"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color rose 1`] = `
<text
  className="ax-word ax-word--rose"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with color teal 1`] = `
<text
  className="ax-word ax-word--teal"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;

exports[`Word renders with defaultProps 1`] = `
<text
  className="ax-word"
  onClick={undefined}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(0,0)"
/>
`;

exports[`Word renders with onClick 1`] = `
<text
  className="ax-word ax-word--clickable"
  onClick={[Function]}
  style={
    Object {
      "animationDelay": "0",
      "fontFamily": undefined,
      "fontSize": undefined,
    }
  }
  textAnchor="middle"
  transform="translate(undefined,undefined)"
/>
`;
